docs(protocol): add inventory exchange (#115)
witnet/witnet-rust: docs/protocol/network/messages/peer-discovery.md

# Peer discovery

> Peer discovery protocol allows nodes to exchange information of known peers to other nodes. This information is usually used by peer discovery algorithms.

Reviewer: "Usually" seems ambiguous. Maybe even ominous.

Author: Ok, I removed it. I left it as: "This information may be used by peer discovery algorithms".

> Nodes may request to their outbound peers to receive a list of their known "recent" peers. This request is initiated by sending a `get_peers` message to a remote peer. The receiving node will reply with a `peers` message with a list of peer addresses that have been seen recently active in the network (e.g. peers that sent at least a message during the last 90 minutes). Usually, the transmitting node will then update the peer addresses into its database

Reviewer 2: ```suggestion
Nodes may request to their outbound peers for a list of their known "recent" peers. This request is initiated by sending a `get_peers` message to a remote peer. The receiving node will reply a `peers` message with a list of peer addresses that have been recently seen active in the network (e.g. peers that sent at least a message in the last 90 minutes). Usually, the transmitting node will then update its local list of peer addresses accordingly.
```

```ascii
         NodeA                            NodeB
           +                                +
           |           GET_PEERS            |
           +------------------------------->+
           |             PEERS              |
           +<-------------------------------+
           |                                |
           +                                +
```

## Get peers message

> The `get_peers` message only consists of the message header with the command `GET_PEERS`.

Reviewer 2: ```suggestion
The `get_peers` message only consists of a message header with the `GET_PEERS` command.
```

## Peers message

> The `peers` message consists of the message header with the command `PEERS` and as payload it contains a list of known peers as:

Reviewer 2: ```suggestion
The `peers` message consists of a message header with the `PEERS` command and a payload containing a list of known peers as:
```

| Field | Type   | Description                                                                          |
| ----- | :----: | ------------------------------------------------------------------------------------ |

> | peers | addr[] | list of IP addresses of active known peers, as described in the [IP address] section |

Reviewer 2: (``) and uppercase.
